src/OrbitGl/BasicPagefaultTrack.cpp

```diff
@@ -0,0 +1,88 @@
+  if (previous_time_and_values_.has_value()) {
+    std::array<double, kBasicPagefaultTrackDimension> differences;
+    std::transform(values.begin(), values.end(), previous_time_and_values_.value().second.begin(),
+                   differences.begin(), std::minus<double>());
+    series_.AddValues(previous_time_and_values_.value().first, differences);
+  }
+
+  previous_time_and_values_ = std::make_pair(timestamp_ns, values);
+}
+
+void BasicPagefaultTrack::AddValuesAndUpdateAnnotations(
+    uint64_t timestamp_ns, const std::array<double, kBasicPagefaultTrackDimension>& values) {
+  AddValues(timestamp_ns, values);
+  SetValueUpperBound(absl::StrFormat("Maximum count: %.0f", GetGraphMaxValue()),
+                     GetGraphMaxValue());
+  SetValueLowerBound(absl::StrFormat("Minimum count: %.0f", GetGraphMinValue()),
+                     GetGraphMinValue());
+}
+
+void BasicPagefaultTrack::SetIndexOfSeriesToHighlight(size_t series_index) {
+  if (series_index >= kBasicPagefaultTrackDimension) return;
+  index_of_series_to_highlight_ = series_index;
+}
+
+void BasicPagefaultTrack::Draw(Batcher& batcher, TextRenderer& text_renderer,
+                               uint64_t current_mouse_time_ns, PickingMode picking_mode,
+                               float z_offset) {
+  LineGraphTrack<kBasicPagefaultTrackDimension>::Draw(batcher, text_renderer, current_mouse_time_ns,
+                                                      picking_mode, z_offset);
+
+  if (picking_mode != PickingMode::kNone || IsCollapsed()) return;
+  AnnotationTrack::DrawAnnotation(batcher, text_renderer, layout_,
+                                  GlCanvas::kZValueTrackText + z_offset);
+}
+
+void BasicPagefaultTrack::DrawSingleSeriesEntry(
+    Batcher* batcher, uint64_t start_tick, uint64_t end_tick,
+    const std::array<float, kBasicPagefaultTrackDimension>& current_normalized_values,
+    const std::array<float, kBasicPagefaultTrackDimension>& next_normalized_values, float z,
+    bool is_last) {
+  LineGraphTrack<kBasicPagefaultTrackDimension>::DrawSingleSeriesEntry(
+      batcher, start_tick, end_tick, current_normalized_values, next_normalized_values, z, is_last);
+
+  if (!index_of_series_to_highlight_.has_value()) return;
+  if (current_normalized_values[index_of_series_to_highlight_.value()] == 0) return;
+
+  const Color kHightlightingColor(231, 68, 53, 100);
+  float x0 = time_graph_->GetWorldFromTick(start_tick);
+  float width = time_graph_->GetWorldFromTick(end_tick) - x0;
+  float content_height =
+      size_[1] - layout_->GetTrackTabHeight() - layout_->GetTrackBottomMargin() - GetLegendHeight();
+  float y0 = pos_[1] - size_[1] + layout_->GetTrackBottomMargin();
+  batcher->AddShadedBox(Vec2(x0, y0), Vec2(width, content_height), z, kHightlightingColor);
+}
+
+bool BasicPagefaultTrack::IsCollapsed() const {
+  return collapse_toggle_->IsCollapsed() || GetParent()->IsCollapsed();
```

Comment:
We need this for each single subtrack. You can collapse only one subtrack and leave the other one expanded.

Reply:
Here I was speaking about duplicated code, I expect this code to be in every kind of subtrack (Gpu, mayor/minor page fault, etc), but now I'm thinking that IsCollapsed() should return only the state of the collapse_toggle and the collapsed state of the parent should determine if the subtracks is being drawn or not. I tend to think that we should try to avoid using GetParent() when possible and instead iterating through the childrens. wdyt?

Reply:
Sorry that I don't have a good solution to avoid using parent track collapsing state in the subtracks. I will leave it as it is for now.